src/Documenter.jl

```diff
@@ -393,66 +393,50 @@ function deploydocs(;
         end
     end
 
-    # Sanity checks
-    if !isempty(travis_repo_slug) && !occursin(travis_repo_slug, repo)
-        @warn("repo $repo does not match $travis_repo_slug")
+    @info "Documenter: checking deployment criteria."
+    should_deploy = true
+    if !occursin(travis_repo_slug, repo)
+        @warn "Documenter: '$(travis_repo_slug)' (TRAVIS_REPO_SLUG) does not occur in '$(repo)' (repo kwarg)."
+        should_deploy = false
     end
-
-    # When should a deploy be attempted?
-    should_deploy =
-        occursin(travis_repo_slug, repo) &&
-        travis_pull_request == "false"   &&
-        (
-            travis_branch == devbranch ||
-            travis_tag    != ""
-        )
-
-    # check that the tag is valid
-    if should_deploy && !isempty(travis_tag) && !occursin(Base.VERSION_REGEX, travis_tag)
-        @warn("tag `$(travis_tag)` is not a valid VersionNumber")
+    if travis_pull_request != "false"
+        @warn "Documenter: '$(travis_pull_request)' (TRAVIS_PULL_REQUEST) should be 'false'."
         should_deploy = false
     end
-
-    # check DOCUMENTER_KEY only if the branch, Julia version etc. check out
-    if should_deploy && isempty(documenter_key)
-        @warn("""
-            DOCUMENTER_KEY environment variable missing, unable to deploy.
-              Note that in Documenter v0.9.0 old deprecated authentication methods were removed.
-              DOCUMENTER_KEY is now the only option. See the documentation for more information.""")
+    if !isempty(travis_tag)
+        if !occursin(Base.VERSION_REGEX, travis_tag)
+            @warn "Documenter: '$(travis_tag)' (TRAVIS_TAG) is not valid VersionNumber."
+            should_deploy = false
+        end
+    end
+    if isempty(travis_tag) && travis_branch != devbranch
+        @warn "Documenter: '$(travis_branch)' (TRAVIS_BRANCH) should match '$(devbranch)' (devbranch kwarg)"
         should_deploy = false
     end
-
-    if get(ENV, "DOCUMENTER_DEBUG", "") == "true"
-        Utilities.debug("TRAVIS_REPO_SLUG       = \"$travis_repo_slug\"")
-        Utilities.debug("  should occur in \"$repo\" (kwarg: repo)")
-        Utilities.debug("TRAVIS_PULL_REQUEST    = \"$travis_pull_request\"")
-        Utilities.debug("  deploying if equal to \"false\"")
-        Utilities.debug("TRAVIS_BRANCH          = \"$travis_branch\"")
-        Utilities.debug("TRAVIS_TAG             = \"$travis_tag\"")
-        Utilities.debug("  deploying if branch equal to \"$devbranch\" (kwarg: devbranch) or tag is set")
-        Utilities.debug("git commit SHA         = $sha")
-        Utilities.debug("DOCUMENTER_KEY exists  = $(!isempty(documenter_key))")
-        Utilities.debug("should_deploy          = $should_deploy")
```

Comment:
Hmm, it might be good if we would still output all the environment variables and their values at the top of `deploydocs` (in addition to the warnings; "Here are all the environment variables affecting deploydocs:").

Reply:
```
┌ Debug: Deployment criteria:
│ - ENV["TRAVIS_REPO_SLUG"]="JuliaLang/julia" occurs in repo="JuliaLang/julia" ✔
│ - ENV["TRAVIS_PULL_REQUEST"]="false" is "false" ✔
│ - ENV["TRAVIS_TAG"]="v0" is (i) empty or (ii) a valid VersionNumber ✔
│ - ENV["TRAVIS_BRANCH"]="master" matches devbranch="master" ✔
│ - ENV["DOCUMENTER_KEY"] exists ✔
│ Deploying: ✔
└ @ Main REPL[32]:1
```

Reply:
Output looks good, but it won't show by default, right? Maybe best to always print this?

Reply:
Make it an `@info`?

Reply:
Yep, for example.

Reply:
It will typically be displayed for every doc-iteration locally, but its pretty short so its probably fine.